tests/unit/test_util.cpp

```diff
@@ -78,3 +78,17 @@ TEST_F(UtilTest, testFromHex)
 
     ASSERT_THROW(result = mococrw::utility::fromHex("  deadbeef"), mococrw::MoCOCrWException);
 }
+
+TEST_F(UtilTest, stringCleanse)
+{
+    std::string secret{"1234"};
+    mococrw::utility::stringCleanse(secret);
+    ASSERT_THAT(secret, Eq(std::string()));
+}
+
+TEST_F(UtilTest, vectorCleanse)
+{
+    std::vector<uint8_t> secret = {222, 173, 190, 239};
+    mococrw::utility::vectorCleanse(secret);
+    ASSERT_THAT(secret, Eq(std::vector<uint8_t>()));
```

Comment:
Since we do not change capacity of the vector inside `vectorCleanse()` you can assert of data was actually overwritten with 0-s. To be on a safe side, you can check object `.capacity()` before  checking for zeros.

BTW: I have noticed that we `.clear()` strings but don't `.clear()` vectors.  This is rather inconsistent.